### DISCUSSION TITLE
Use walletconnect without a modal

### DISCUSSION THREAD

**Discussion by adrianotadao** [WalletConnect/walletconnect-monorepo#1673] [Developer Support] [answered]
Hello folks, 

The modal style doesn't fit well with my application so I was wondering if there is a way of using walletconnect without a modal. Is it possible?

thank you!

**Comment by xzilja** [comment, marked as answer]
Yep, you are looking for plain sign sdk in this case https://docs.walletconnect.com/2.0/introduction/sign

**Reply by adrianotadao** [reply]
@IljaDaderko I see. Have you seen a good example doing it, by any chance? All the examples use the same modal and this is not clear to me how this would work. Thank you in advance

**Reply by xzilja** [reply]
On walletconnect end, you can reference sign sdk docs https://docs.walletconnect.com/2.0/javascript/sign/installation 
There is an example with `const { uri, approval } = await signClient.connect( ... )` which passes uri to qrcode modal, this is where you'd replace it with your own ui implementation.

You essentially need to take that uri and generate qrcode with it / create linking that passes it to native devices.
Not sure if there are any good examples of this in the wild, but you can always look at web3modal source code.

As a side note, we added few customisation options to v2 modal that you can play with here https://web3modal-dev.pages.dev perhaps you can find something that fits your ui. We'll be adding more theming options soon as well. What exactly would make it fit in your design better?

**Reply by adrianotadao** [reply]
thank you very much for answering me, @IljaDaderko!

> What exactly would make it fit in your design better?

The biggest problem for me is the modal existence. I would love to be able to have something magic like 

```javascript
import WalletConnectPanel from '@walletconnect/magic-ui'

render (
  <div id='walletconnect'>
    <WalletConnectPanel config={...config} />
  </div>
)
```
where `<WalletConnectPanel />` would simply render the **qrcode**, **the available wallets** and so on and then allow me to apply my styles on it. That would be just perfect.

**Reply by vildantursic** [reply]
@adrianotadao I don't know if forking Web3Modal would work for you and modifying styles to fit your need. Moving wallets from modal to something else. I was planning for doing something similar but I have tight deadlines so no time for that at the moment 😄

**Reply by adrianotadao** [reply]
@vildantursic we had the same idea here and we came to the same conclusion that we would have no time for maintaining it. :laughing: 

btw I'm very surprised how web3 in general likes those modals, omg.
